cmake/SetupHdf5.cmake

```diff
@@ -38,7 +38,11 @@ endif()
 # are listed here:
 # https://github.com/HDFGroup/hdf5/blob/develop/doc/file-locking.md
 include(CheckCXXSourceCompiles)
-set(CMAKE_REQUIRED_LIBRARIES hdf5::hdf5)
+if(HDF5_USE_STATIC_LIBRARIES)
+  set(CMAKE_REQUIRED_LIBRARIES ${HDF5_C_LIBRARIES})
```

Comment:
This will break if the HDF5 headers are not in some global include path. You can set `CMAKE_REQUIRED_INCLUDES` as well I think: https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html.
Why is this necessary? Does the `hdf5::hdf5` target carry the wrong (non-static) libs?

Reply:
Something like that must be the case yes. It failed to link against libz when I used static libs and only did `hdf5::hdf5`

Reply:
Hm then it should fail in our executables too. My guess is that in the executables zlib is pulled in from charm instead, which also links zlib. Here's an upstream issue: https://forum.hdfgroup.org/t/cmake-zlib-dependency-not-properly-propagted-by-hdf5-targets/5149/8

Reply:
Okay, I'll think about this a bit to see what I can think of in terms of finding a portable solution that doesn't require Charm++ to link in libz. Thanks for finding that issue!

Reply:
Aha, if I link against `HDF5::HDF5` it works but not if I do `hdf5::hdf5`